src/main/java/net/javadiscord/javabot/systems/moderation/AutoMod.java

```diff
@@ -121,25 +140,32 @@ private void checkContentAutomod(@Nonnull Message message) {
 					);
 			message.delete().queue();
 		}
+
 		final String messageRaw = message.getContentRaw();
+		Matcher urlMatcher = urlPattern.matcher(messageRaw);
 		if (messageRaw.contains("http://") || messageRaw.contains("https://")) {
 			// only do it for a links, so it won't iterate for each message
-			for (String spamUrl : spamUrls) {
-				if (messageRaw.contains(spamUrl)) {
-					try {
-						new ModerationService(message.getJDA(), Bot.config.get(message.getGuild()).getModeration())
-								.warn(
-										message.getMember(),
-										WarnSeverity.HIGH,
-										"Automod: Suspicious Link",
-										message.getGuild().getMember(message.getJDA().getSelfUser()),
-										message.getTextChannel(),
-										false
-								);
-						message.delete().queue();
-					} catch (Exception e) {
-						e.printStackTrace();
+			while (urlMatcher.find()) {
+				String url = messageRaw.substring(urlMatcher.start(1), urlMatcher.end());
```

Comment:
maybe use `urlMatcher.group(1)` instead of `messageRaw.substring(urlMatcher.start(1), urlMatcher.end())` here.

Reply:
won't it only return first url?

Reply:
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Matcher.html#group(int)
> Returns the input subsequence **captured by the given group** during the previous match operation.